### PR TITLE
Switch Luna tracing to use materialize.

### PR DESCRIPTION
**Pull request by mdwelsh**
This PR migrates Luna's execution tracing to use the new materialize support.

### CODE REVIEW

File: lib/sycamore/sycamore/query/execution/sycamore_executor.py [hunk 1]
```diff
@@ -181,22 +178,24 @@ def process_node(self, logical_node: LogicalOperator, query_id: str) -> Any:
...
+
+        if self.dry_run:
+            code, imports = operation.script()
+            self.imports += imports
+            self.node_id_to_code[logical_node.node_id] = code
+            self.node_id_to_node[logical_node.node_id] = logical_node
+            result = None
+        else:
+            result = operation.execute()
+            if trace_dir and hasattr(result, "materialize"):
```
Comment (tranade) [inline lib/sycamore/sycamore/query/execution/sycamore_executor.py:196]: do all docsets have materialize? wondering if we can get this to write out traces after a top k operation

File: apps/query-ui/queryui/queryui.py [hunk 2]
```diff
@@ -46,34 +46,40 @@ def show_schema(container: Any, schema: Dict[str, Tuple[str, Set[str]]]):
...
-                        if p not in BASE_PROPS and p in doc.properties:
-                            doc_list[p] = doc.properties[p]
-                    doc_list["text_representation"] = doc.text_representation
-                    data_list.append(doc_list)
+                    # For now, skip over MetadataDocuments.
+                    if "doc_id" not in doc:
+                        continue
+                    data_list.append(doc)
 
         df = pd.DataFrame(data_list)
```
Comment (tranade) [inline apps/query-ui/queryui/queryui.py:64]: is it possible to let each doc.properties and doc.properties.entity have its own column? helps when debugging so you can tell if something is wrong from a quick glance

File: lib/sycamore/sycamore/query/execution/sycamore_executor.py (hunk 1 above)
Comment (mdwelsh) [reply lib/sycamore/sycamore/query/execution/sycamore_executor.py:196]: Yes, it is a top-level docset operation.

File: apps/query-ui/queryui/queryui.py (hunk 2 above)
Comment (mdwelsh) [reply apps/query-ui/queryui/queryui.py:64]: Yes, we can tweak it, but I am working on a new UI for this anyway.

File: lib/sycamore/sycamore/query/execution/sycamore_executor.py (hunk 1 above)
Comment (tranade) [reply lib/sycamore/sycamore/query/execution/sycamore_executor.py:196]: Hmm, it isn't writing out traces for a docset after top k (when I checked locally)

**Review by tranade — APPROVED** [review]

**Pull request merged**